### PR TITLE
bugfix/#4164/The_error_messages_are_absent_for_'Phone'_field_when_Admin_enters_invalid_values_on_'Information_about_client'_page

### PR DESCRIPTION
**Pull request by SokhaTaras**
Updated the behavior of warning fields
Points 3 and 4 cannot be reproduced, they are already working correctly. I guess, it was fixed earlier

### CODE REVIEW

**Review by zaitsev-oleksii — COMMENTED** [review]
1. I suggest using `app/shared/ubs-input-error` component to display error: onChanges will trigger on component input changes, functions in template - on every detection change;
2. Use `app/shared/phone-validator` instead of `pattern validator`

File: src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.ts [hunk 1]
```diff
@@ -63,6 +64,16 @@ export class UbsAdminOrderClientInfoComponent implements OnInit, OnDestroy {
...
+  getErrorMessage(abstractControl: AbstractControl) {
+    if (abstractControl.errors.required) {
+      return 'input-error.required';
+    }
+
+    if (abstractControl.errors.pattern) {
+      return 'input-error.number-length';
+    }
+  }
+
```
Comment (zaitsev-oleksii) [inline src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.ts:76]:
```suggestion
```

File: src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.html [hunk 2]
```diff
@@ -64,6 +64,11 @@ <h3 class="form__headline">{{ 'customer-info.title' | translate }}</h3>
               [imask]="{ mask: phoneMask }"
             />
           </div>
+          <div *ngIf="userInfoDto.controls['recipientPhoneNumber'].invalid && userInfoDto.controls['recipientPhoneNumber'].touched">
+            <div class="error-message">
+              {{ getErrorMessage(userInfoDto.controls['recipientPhoneNumber']) | translate }}
+            </div>
+          </div>
```
Comment (zaitsev-oleksii) [inline src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.html:71]:
```suggestion
          <div *ngIf="userInfoDto.controls['recipientPhoneNumber'].invalid && userInfoDto.controls['recipientPhoneNumber'].touched">
            <div class="error-message">
              <app-ubs-input-error
                [formElement]="userInfoDto.controls['recipientPhoneNumber']"
                [phoneNumberFieldValue]="userInfoDto.controls['recipientPhoneNumber'].value"
              ></app-ubs-input-error>
            </div>
          </div>
```

File: src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import { Component, Input, OnDestroy, OnInit } from '@angular/core';
-import { FormGroup } from '@angular/forms';
+import { FormGroup, AbstractControl } from '@angular/forms';
```
Comment (zaitsev-oleksii) [inline src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.ts:2]:
```suggestion
import { FormGroup } from '@angular/forms';
```

**Review by zaitsev-oleksii — CHANGES_REQUESTED** [review]
Change validators for `recipientPhoneNumber` in `ubs/ubs-admin/components/ubs-admin-order.component`: `Validators.pattern(Patterns.adminPhone)` -> `PhoneNumberValidator('UA')`;
If there is a need for `wrong number length` error message,  create custom validator for it.

File: src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.html (hunk 2 above)
Comment (SokhaTaras) [reply src/app/ubs/ubs-admin/components/ubs-admin-order-client-info/ubs-admin-order-client-info.component.html:71]: You have good suggestions, but, I guess they all are out of scope of this task.

**Review by marianaharnyk — APPROVED** [review]

**Review by NewArtMask — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Review by zaitsev-oleksii — APPROVED** [review]

**Review by voronkooleksandr — APPROVED** [review]

**Pull request merged**